Add "movie" mode to temporal controller

nyalldawson opened the pull request:
This mode is purely for animated map movies -- the user can set
the total number of frames, and the animation will progress
frame by frame for that number of frames, advancing the current
map settings frame at each step (also the @frame_number expression
variable).

No time based filtering of data is performed when in this mode.

nyalldawson committed:
[feature] Add "movie" mode to temporal controller

This mode is purely for animated map movies -- the user can set
the total number of frames, and the animation will progress
frame by frame for that number of frames, advancing the current
map settings frame at each step (also the @frame_number expression
variable).

No time based filtering of data is performed when in this mode.

nyalldawson committed:
Move temporal enums to Qgis, promote to enum class

nyalldawson commented:
Att @timlinux

nirvn reviewed (commented):
@nyalldawson , good looks good. I do wonder whether we should avoid adding a whole page in the temporal controller widget .ui for movie and instead share the navigation one for movie, with some widget->setVisible( true/false ) and widget->setTooltip( str ) to customize the view. We end up duplicating a lot of buttons otherwise.

nyalldawson commented on src/gui/qgstemporalcontrollerwidget.cpp:
```diff
@@ -225,17 +253,26 @@ void QgsTemporalControllerWidget::togglePlayForward()
     mBackButton->setChecked( false );
     mForwardButton->setChecked( true );
+
+    mStopButton_2->setChecked( false );
+    mBackButton_2->setChecked( false );
+    mForwardButton_2->setChecked( true );
```
I agree it's gross. But I don't particularly like the hiding approach either -- when we do that we end up with slightly misspaced layouts, because qt still accounts for the inter-widget spacing even for hidden widgets. So we end up with double / triple / ... spacing instead of single. It's ok in some contexts, but looks bad here :P 

And ultimately we end up with just as complex logic about hiding /showing / changing behaviour of existing widgets. :man_shrugging:

nyalldawson replied:
@nirvn -- ok, i've cleaned this up, and moved the button strip out to a reusable separate widget.

nirvn replied:
@nyalldawson , nice!

nyalldawson committed:
Rename buttons for clarity

nyalldawson committed:
Move playback controller buttons to reusable widget

Create QgsPlaybackControllerWidget as reusable
play/pause/rewind/next/... button bar

nyalldawson committed:
Add icon for movie mode
